### DISCUSSION TITLE
Is it possible to use the data returned from a mutation and not re-run the queries?

### DISCUSSION THREAD

**Discussion by AlecRobertson** [urql-graphql/urql#2809] [Q&A] [answered]
First of all thank you for your superb work on this project!

I have a query which gets some data, let's say it looks like this:

```
query {
  getData(ID: 1234) {
    ID
    someData {
        ID
        something
    }
  }
}
```

I use "someData" in a component which can call a mutation which updates "someData" and returns the updated "someData". When this happens, the query getData re-runs and I get the updated "someData". But I already have the updated "someData" from the mutation, is there a way to therefore disable re-running the query and just use the data that comes back from the mutation?

**Comment by JoviDeCroock** [comment, marked as answer]
Thank you for the kind words and welcome to the urql community!

There is a way to do this by means of the [normalized cache](https://formidable.com/open-source/urql/docs/graphcache/) by default urql uses a document based cache which means we'll map your query to the response but we don't store a lot of metadata hence we have no choice but to refetch. With the normalized cache we automatically update the entity
